clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -104,12 +136,33 @@ func (c *clusterClient) Apply(manifest string) error {
 	return c.waitForKubectlApply(manifest)
 }
 
-func (c *clusterClient) GetClusterObjects() ([]*clusterv1.Cluster, error) {
+func (c *clusterClient) GetContextNamespace() string {
+	if c.configOverrides.Context.Namespace == "" {
+		c.configOverrides.Context.Namespace = apiv1.NamespaceDefault
```

Comment:
It isn't clear to me why this needs to set the context namespace vs. just returning the default namespace. Can you explain why it's doing that?

Reply:
The intention for setting the context was to save the work of figuring out what context to use.
But I agree because the method is called `GetContextNamespace`, it should just return the default namespace.